src/app/pokemon-list/pokemon-list.component.ts

```diff
@@ -10,29 +16,53 @@ import {
   POKEMON_NUMBER,
 } from '../_core/constants/pagination.constants';
 import { PokemonSpeciesList } from '../_core/models/pokemon';
+import { PokemonErrorComponent } from '../pokemon-error/pokemon-error.component';
 
 @Component({
   selector: 'app-pokemon-list',
-  imports: [CommonModule, MatPaginatorModule, RouterModule],
+  imports: [
+    CommonModule,
+    MatPaginatorModule,
+    RouterModule,
+    PokemonErrorComponent,
+  ],
   templateUrl: './pokemon-list.component.html',
 })
 export class PokemonListComponent {
   readonly page = input<number>(DEFAULT_PAGE);
   readonly limit = input<number>(LIMIT_PAGE);
-  pokemonNumber: number = POKEMON_NUMBER;
+  POKEMON_NUMBER: number = POKEMON_NUMBER;
   PAGE_SIZE_OPTIONS: number[] = PAGE_SIZE_OPTIONS;
 
   readonly offset: Signal<number> = computed(() => this.page() * this.limit());
 
-  readonly pokemons: HttpResourceRef<PokemonSpeciesList | undefined> =
+  readonly allPokemons: HttpResourceRef<PokemonSpeciesList | undefined> =
     httpResource(() => ({
       url: `https://pokeapi.co/api/v2/pokemon-species`,
       params: {
-        limit: this.limit(),
-        offset: this.offset(),
+        limit: this.POKEMON_NUMBER,
       },
     }));
 
+  readonly search = input<string | undefined>();
+
+  readonly filteredPokemons = linkedSignal(() => {
+    const list = this.allPokemons.value()?.results ?? [];
+    const q = this.search()?.trim().toLowerCase();
+
+    if (!q) return list;
+
+    return list.filter((p) => p.name.toLowerCase().includes(q));
+  });
+
+  readonly pokemons = computed(() => {
+    const all = this.filteredPokemons();
+    const offset = this.offset();
+    const limit = Number(this.limit());
+
+    return all.slice(offset, offset + limit);
+  });
```

Comment:
YOu don't need this. 

You only need your allPokemon signals and the filteredPokemon computed signal

For now we don't do a search on the backend only a filter on the UI on the current elements of the page